### ISSUE TITLE
[BUG] wash ctl returns successful exit code even if it can't connect to the cluster

### ISSUE THREAD

**Issue by alsuren** [wasmCloud/wash-archive#186]
**Describe the bug**
wash ctl returns successful exit code even if it can't connect to the cluster

If you don't have nats running, and you do a wash ctl command then you get an error message:
```
$ wash ctl link put \
    MBCFOPM6JW2APJLXJD3Z5O4CN7CPYJ2B4FTKLJUR5YR5MITIU7HD3WD5 \
    VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M \
    wasmcloud:httpserver \
    ADDRESS=0.0.0.0:8081
...
Error advertising link: Connection refused (os error 61)
```
but then the process exit code (`$?` in bash) is 0 (success).

**Expected behavior**
If an error occurs, I would expect `wash` to return a nonzero exit code.

**Screenshots**

**Environment (please complete the following information) **
 - OS: MacOS
 - Shell bash
 - wash Version 0.6.6

**Additional context**
This behaviour is important when writing bash scripts (especially when using `set -e` to make the script exit early when a command fails)

**Comment by brooksmtownsend** [comment]
For `ctl` commands, if the NATS connection fails with `Connection refused` that should be bubbled up to the main logic to return as an error, not as an `Ok` result.

**Comment by brooksmtownsend** [comment]
This may be addressed by #221, further verification needed

**Comment by brooksmtownsend** [comment]
<img width="860" alt="image" src="https://user-images.githubusercontent.com/12040685/164277613-2aabbeb1-90d4-4c69-9869-6a4a9b9b75f8.png">
This is no longer an issue, likely fixed with the refactor to anyhow errors.
